hydra-task/src/main/java/com/addthis/hydra/task/map/StreamMapper.java

```diff
@@ -366,4 +370,9 @@ private void maybeCloseJmx() {
             }
         }
     }
+
+    public CompletableFuture<Void> getOnTaskComplete() {
```

Comment:
Maybe we should call it completionFuture or something. "onTaskComplete" more describes a specific use case (adding hooks) than what it is. It might be kinda awkward looking to do `task.getOnTaskComplete().join()` for instance.